src/main/java/redis/clients/jedis/Pipeline.java

```diff
@@ -20,8 +20,8 @@
       List<Object> values = new ArrayList<>();
 
       if (list.size() != responses.size()) {
-        throw new JedisDataException("Expected data size " + responses.size() + " but was "
-            + list.size());
+        throw new IllegalStateException(
```

Comment:
One could argue whether `JedisDataException` is more appropriate than `IllegalStateException`. In any case, there's no way to recover from here as that case could only be a result of a bug within Jedis or Redis.

Reply:
We[*] are trying to dedicate `JedisDataException` only for a direct response from Redis, i.e. `-ERR` response. From that perspective, it's not valid here.

[*] Frankly, it's just me right now. But the idea was generated and supported way back by both @HeartSaVioR and @marcosnils. But it is always getting stuck because of the backward incompatibility.